### PR TITLE
Draft PR for Curve-Amm

### PR DESCRIPTION
**Pull request by vivekvpandya**
Resolves # .

-
-
-

@reviewer-or-team

### CODE REVIEW

File: examples/curve-amm/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+            .ok_or(CurveAmmError::Math)?;
+        let old_balances = pool.balances.clone();
+        let d0 = Self::get_d(&self, &old_balances, ann).ok_or(CurveAmmError::Math)?;
+        let mut token_supply = FixedU128::zero();
+        let reply = msg::send_and_wait_for_reply(
+            pool.pool_asset,
+            &Action::TotalIssuance.encode(),
+            1_000_000_000,
+            0,
+        )
```
Comment (breathx) [inline examples/curve-amm/src/lib.rs:464]:
```suggestion
        let reply: Event = msg::send_and_wait_for_reply(
            pool.pool_asset,
            Action::TotalIssuance,
            1_000_000_000,
            0,
        )
```

**Comment by NikVolf** [comment]
I think it's time to setup separate repo for this

**Comment by shamilsan** [comment]
Please move this PR here: https://github.com/gear-tech/apps

**Comment by vivekvpandya** [comment]
> Please move this PR here: https://github.com/gear-tech/apps

How to test apps in this new repo?

**Comment by shamilsan** [comment]
> How to test apps in this new repo?

We will add gtest/RPC-based checks a little bit later.  One should test programs locally at the moment.

**Comment by shamilsan** [comment]
> How to test apps in this new repo?

Just added checking with `gtest`: https://github.com/gear-tech/apps/pull/2

**Comment by shamilsan** [comment]
Superseded by https://github.com/gear-tech/apps/pull/7

**Comment by vivekvpandya** [comment]
Yes this can be closed.

**Pull request closed without merging**